services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/histogram_builder.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Research
+    module GenAI
+      class HistogramBuilder < ApplicationService
+        attr_reader :scores
+
+        def initialize(scores:)
+          @scores = scores
+        end
+
+        def run
```

Comment:
Consider using [tally](https://ruby-doc.org/3.3.1/Enumerable.html#method-i-tally)
```ruby
def run = scores.map(&:round).tally

Reply:
😮

Reply:
`HistogramBuilder` is history